File: 22237205/src/main/java/org/example/Motorcycle.java
package org.example;

class Motorcycle extends com.vehicleRental.Vehicle {
    private boolean hasSidecar;

    public Motorcycle(String vehicleId, String model, double baseRentalRate, boolean hasSidecar) {
        super(vehicleId, model, baseRentalRate);
        this.hasSidecar = hasSidecar;
    }

    @Override
    public double calculateRentalCost(int days) {
        return getBaseRentalRate() * days * (hasSidecar ? 1.2 : 1.0);
    }

    @Override
    public boolean isAvailableForRental() {
        return isAvailable();
    }

    @Override
    public String toString() {
        return "Motorcycle{" + "model='" + getModel() + '\'' + ", hasSidecar=" + hasSidecar + '}';
    }
}

File: Rental System 22237205/src/main/kotlin/Motorcycle.kt
package com.example.vehicle

import Vehicle

class Motorcycle(
    vehicleId: String,
    model: String,
    baseRentalRate: Double,
    private val hasSidecar: Boolean
) : Vehicle(vehicleId, model, baseRentalRate) {

    override fun calculateRentalCost(days: Int): Double {
        return if (hasSidecar) getBaseRentalRate() * 1.2 * days else getBaseRentalRate() * days
    }

    override fun isAvailableForRental(): Boolean {
        return isAvailable()
    }

    override fun toString(): String {
        return "Motorcycle(model='${getModel()}', sidecar=$hasSidecar)"
    }
}